### ISSUE TITLE
Top bar 2.0

### ISSUE THREAD

**Issue by bouncepaw** [bouncepaw/mycorrhiza#78]
No one likes the current top bar.

Let's address what is wrong with it now:
* Looks sad when it gets multiline, especially on mobile devices.
* It is hard to make all the entries fit one line _along with_ user-related links (_u_ hypha or login/register) **and** search bar.

What I want top bar 2.0 to have:
* Meaningful responsive design
* Everything fits
* Useful for navigation

## Two-line approach
Here's what I though of. We have two lines:

1. Link to the home hypha, search bar, user links. I'm sure there is a way to fit them all on any computer screen.
2. Navigation links. More advanced than now, dropdowns supported maybe.

    The dropdowns should work without JS. We have the tag for that: `<details>`.

On mobile devices, it may look like that (top to bottom):
1. Home hypha link
2. Search bar
3. Uh some collapsible menu
    1. User links
    2. The navigation links. Any collapsibles inside should be uncollapsed beforehand I guess.

### Default navigation links
If we take this approach, we will have much more space for the navigation links. What do we put there? Not much, I guess.
1. Recent changes
2. List
3. Random
4. Admin panel, for admins only

### Configuration
The current approach with the special hypha shall be demolished and forgotten. Perhaps.

## The sidebar approach
What if we take all the navigation links and put them into a sidebar to the left of the hypha section?

Well, it is possible, but I wanted to put TOC there...

**Comment by bouncepaw** [comment]
No one really liked the two-line approach, so it seems, but it will go to 1.3 nevertheless.

I may try to attempt to try to implement something else later, perhaps.

**Comment by bouncepaw** [comment]
With localization added in 1.5, the top bar 2.0 should support multiple languages too

**Comment by chekoopa** [comment]
> With localization added in 1.5, the top bar 2.0 should support multiple languages too

...And that would mean **wiki-local l18n support**. Oh boy!

**Comment by bouncepaw** [comment]
Once #115 is done and we have the word Profile in our vocabulary, we can replace the current user's name in the topbar with the word Profile. That would fix the overflow problem with long names (Yogurt has complained about that once or twice).

Also, once #115 is done, there is no need to have the admin panel link in the top bar. We will just remove it.

Next, we can have URL `/your-profile` redirect to your profile. We can put this link into the default top bar link list (which is _Recent changes, All hyphae, Random, Help_ now, by the way) and get rid of the special zone for the user links at all.

Then we can think of a better layout.

**Comment by handlerug** [comment]
No, we should not remove the username from the header. It's important for the context.

**Comment by bouncepaw** [comment]
Why not? What context? Isn't the wiki name and that emoji in the title not enough for the context?

**Comment by handlerug** [comment]
That's not it. I like having my username displayed in the header at a glance, because I have different usernames across different wikis.

**Comment by bouncepaw** [comment]
Fumu fumu

**Comment by bouncepaw** [comment]
A possible redesign. The top bar becomes a sidebar in a wide viewport, like on the screenshot below. It will be possible once we get rid of the sibling sidebar for real, if that will ever happen.

<img width="1392" alt="Снимок экрана 2022-04-21 в 18 28 41" src="https://user-images.githubusercontent.com/22562024/164497043-1c79d531-938e-4335-b179-667c3d585bcc.png">

On two-column and one-column layouts, what will happen? Dunno. Just an idea

**Comment by bouncepaw** [comment]
I said,

> The current approach with the special hypha shall be demolished and forgotten. Perhaps.

Now I say no “perhaps”. The special hypha has to go.

Also considering the cool navigation inspired by Betula. You know, horizontal scrolling on phones and whatnot.
